src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByAuthorizationCodeParameterBuilder.cs

```diff
@@ -101,5 +102,26 @@ public AcquireTokenByAuthorizationCodeParameterBuilder WithPkceCodeVerifier(stri
             Parameters.PkceCodeVerifier = pkceCodeVerifier;
             return this;
         }
+
+        /// <summary>
+        /// To help with resiliency, AAD Cached Credential Service (CCS) operates as an AAD backup.
+        /// </summary>
+        /// <param name="userObjectIdentifier">GUID which is unique to the user, parsed from the client_info.</param>
+        /// <param name="tenantIdentifier">GUID format of the tenant ID, parsed from the client_info.</param>
+        /// <returns>The builder to chain the .With methods</returns>
+        public AcquireTokenByAuthorizationCodeParameterBuilder WithCcsRoutingHint(string userObjectIdentifier, string tenantIdentifier)
```

Comment:
What about UPN routing hint? `WithCcsRoutingHint(string upn)`

Reply:
the other ones would use .WithLoginHint() if it's passed in. no need to add UPN if LoginHint is not available on that builder.

Reply:
Good point.